geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -7767,6 +7768,16 @@ protected void postDestroyRegion(boolean destroyDiskRegion, RegionEventImpl even
         cache.getInternalDistributedSystem().getDistributedMember(), null, op.isClose());
   }
 
+  private void closeDataStoreStats() {
+    PartitionedRegionDataStore localDataStore = dataStore;
```

Comment:
Is this local variable necessary? Could we just use `dataStore` directly in this method?

Reply:
I think it is needed when reading a field that is not final. I know we currently only ever set it to non-null but I just wanted this code to be future proof against any possible future changes that could modify this non-final field.

Reply:
But as a look at all the existing code in this class that simply checks if "dataStore" is null, I'll change this code to also follow that pattern.